pkg/tso/keyspace_group_manager.go

```diff
@@ -415,6 +462,8 @@ func (kgm *KeyspaceGroupManager) watchKeyspaceGroupsMetaChange(revision int64) (
 				case clientv3.EventTypeDelete:
 					kgm.deleteKeyspaceGroup(id)
 				}
+
+				kgm.storageChangeEventsApplied.Add(1)
```

Comment:
Could the counter overflow? Do you think we can only add it during the test? Maybe a failpoint could help.

Reply:
I already changed before saw your comment. Basically, I required read lock in the test. Check the commit 5.